demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,524 @@
+                            const char * pPath )
+{
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+
+    /* Initialize the request object. */
+    requestInfo.pHost = pHost;
+    requestInfo.hostLen = strlen( pHost );
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = strlen( pMethod );
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. */
+    requestInfo.flags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                      &requestInfo );
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        /* Initialize the response object. The same buffer used for storing
+         * request headers is reused here. */
+        response.pBuffer = userBuffer;
+        response.bufferLen = USER_BUFFER_LENGTH;
+
+        LogInfo( ( "Sending HTTP %s request to %s%s...",
+                   pMethod, SERVER_HOST, pPath ) );
+        LogInfo( ( "Request Headers:\n%.*s",
+                   ( int32_t ) requestHeaders.headersLen,
+                   ( char * ) requestHeaders.pBuffer ) );
+        LogInfo( ( "Request Body:\n%.*s\n",
+                   ( int32_t ) REQUEST_BODY_TEXT_LENGTH,
+                   REQUEST_BODY_TEXT ) );
+        /* Send the request and receive the response. */
+        httpStatus = HTTPClient_Send( pTransportInterface,
+                                      &requestHeaders,
+                                      ( uint8_t * ) REQUEST_BODY_TEXT,
+                                      REQUEST_BODY_TEXT_LENGTH,
+                                      &response,
+                                      0 );
+    }
+    else
+    {
+        LogError( ( "Failed to initialize HTTP request headers: Error=%s.",
+                    HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Received HTTP response from %s%s...",
+                   SERVER_HOST, pPath ) );
+        LogInfo( ( "Response Headers:\n%.*s",
+                   ( int32_t ) response.headersLen,
+                   response.pHeaders ) );
+        LogInfo( ( "Response Status:\n%u",
+                   response.statusCode ) );
+        LogInfo( ( "Response Body:\n%.*s\n",
+                   ( int32_t ) response.bodyLen,
+                   response.pBody ) );
+    }
+    else
+    {
+        LogError( ( "Failed to send HTTP %s request to %s%s: Error=%s.",
+                    pMethod, SERVER_HOST, pPath, HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        return EXIT_SUCCESS;
+    }
+    else
+    {
+        return EXIT_FAILURE;
+    }
```

Comment:
I'm not sure if we're following MISRA for demos as well (I saw an empty else clause above) but if we are then we shouldn't have multiple returns either:

```suggestion
    return ( httpStatus == HTTP_SUCCESS )? EXIT_SUCCESS : EXIT_FAILURE;
```

Reply:
I declared an extra returnStatus and added the following:
```c
if( httpStatus != HTTP_SUCCESS ){
        returnStatus = EXIT_FAILURE;
}
```